eng/vmr-sync.sh

```diff
@@ -165,14 +168,17 @@ dotnet="$scriptroot/../.dotnet/dotnet"
 "$dotnet" tool restore
 
 # Run the sync
-target_sha=$(git -C "$installer_dir" rev-parse HEAD)
-highlight "Starting the synchronization to $target_sha.."
+if [[ -z "$target_ref" ]]; then
+  target_ref=$(git -C "$installer_dir" rev-parse HEAD)
+fi
+
+highlight "Starting the synchronization to $target_ref.."
```

Comment:
Dev UX request - when printing paths in error / debug messages, it's very useful to enclose the variables in `''` (e.g., `highlight "Starting the synchronization to '$target_ref'.."`) so that it's clear what the path is. Especially in this case, when the `$target_ref` variable would be empty (the most common bug in bash scripts), you'd get a perfectly valid, but very misleading message of "Starting the synchronization to ..". I'd like to have this convention adopted in all of our scripts as it makes the messages more clear.